### DISCUSSION TITLE
x509: certificate signed by unknown authority

### DISCUSSION THREAD

**Discussion by toddysm** [mitmproxy/mitmproxy#5570] [Intercept Everything!]
Trying to use mitmproxy to look at the requests sent from docker client to DockerHub registry but after configuring everything according to the docs https://docs.mitmproxy.org/stable/overview-getting-started/#configure-your-browser-or-device I am still getting the following error:

```
Error response from daemon: Get "https://registry-1.docker.io/v2/": x509: certificate signed by unknown authority
```
Any ideas how to make the mitmproxy cert trusted systemwide?

**Comment by Prinzhorn** [comment]
Does this help (e.g. `service docker restart`)? https://stackoverflow.com/questions/50768317/docker-pull-certificate-signed-by-unknown-authority

**Comment by toddysm** [comment]
Well, I am not sure because I see the same problems for websites too. It does seem that the cert is not trusted because Firefox also shows the following:
![image](https://user-images.githubusercontent.com/1007103/189805806-c9c17cce-8ef9-40a3-a15d-31dee4946202.png)

Looks to me that following the instructions did not install the cert properly.

**Reply by toddysm** [reply]
Never mind that last one. It seems I missed the part that FF doesn't use the machine certs and I used https://localhost:8080 for the proxy config in Docker instead http only. Now things work as expected. We can close this one.
